test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
+        "Plottable requires a valid SVG to renderTo", "rejects empty d3 selections");
+      c.destroy();
       svg.remove();
     });
 
-    it("originToSVG() (top-level component)", () => {
-      TestMethods.fixComponentSize(c, cWidth, cHeight);
-      c.renderTo(svg);
+    it("detaches the component if rendering to a new svg", () => {
+      let svgHeight2 = 50;
+      let svg2 = TestMethods.generateSVG(300, svgHeight2);
 
-      c.xAlignment("left").yAlignment("top");
-      let origin = c.originToSVG();
-      assert.strictEqual(origin.x, 0, "returns correct value (xAlign left)");
-      assert.strictEqual(origin.y, 0, "returns correct value (yAlign top)");
+      c.renderTo(svg);
+      assert.isTrue((<Node> svg.node()).hasChildNodes(), "anchored onto svg");
+      assert.strictEqual(c.height(), SVG_HEIGHT, "occupies entire space of svg");
 
-      c.xAlignment("center").yAlignment("center");
-      origin = c.originToSVG();
-      assert.strictEqual(origin.x, (SVG_WIDTH - cWidth) / 2, "returns correct value (xAlign center)");
-      assert.strictEqual(origin.y, (SVG_HEIGHT - cHeight) / 2, "returns correct value (yAlign center)");
+      c.renderTo(svg2);
 
-      c.xAlignment("right").yAlignment("bottom");
-      origin = c.originToSVG();
-      assert.strictEqual(origin.x, SVG_WIDTH - cWidth, "returns correct value (xAlign right)");
-      assert.strictEqual(origin.y, SVG_HEIGHT - cHeight, "returns correct value (yAlign bottom)");
+      assert.isFalse((<Node> svg.node()).hasChildNodes(), "removed from svg");
+      assert.isTrue((<Node> svg2.node()).hasChildNodes(), "anchored onto second svg");
+      assert.strictEqual(c.height(), svgHeight2, "occupies entire space of second svg");
 
+      c.destroy();
+      svg2.remove();
       svg.remove();
     });
+  });
 
-    it("originToSVG() (nested)", () => {
-      TestMethods.fixComponentSize(c, cWidth, cHeight);
-      let group = new Plottable.Components.Group([c]);
-      group.renderTo(svg);
-
-      let groupWidth = group.width();
-      let groupHeight = group.height();
-
-      c.xAlignment("left").yAlignment("top");
-      let origin = c.originToSVG();
-      assert.strictEqual(origin.x, 0, "returns correct value (xAlign left)");
-      assert.strictEqual(origin.y, 0, "returns correct value (yAlign top)");
-
-      c.xAlignment("center").yAlignment("center");
-      origin = c.originToSVG();
-      assert.strictEqual(origin.x, (groupWidth - cWidth) / 2, "returns correct value (xAlign center)");
-      assert.strictEqual(origin.y, (groupHeight - cHeight) / 2, "returns correct value (yAlign center)");
+  it("does not allow the origin object to be modified", () => {
+    c.renderTo(svg);
+    let receivedOrigin = c.origin();
+    let delta = 10;
+    receivedOrigin.x += delta;
+    receivedOrigin.y += delta;
+    assert.notStrictEqual(receivedOrigin.x, c.origin().x, "can be modified without affecting origin (x)");
```

Comment:
change the order of `receivedOrigin.x` and `c.origin().x`
and the message should be *cannot* modify... ?